moveit_ros/benchmarks/src/BenchmarkExecutor.cpp

```diff
@@ -897,6 +919,121 @@ void BenchmarkExecutor::collectMetrics(PlannerRunData& metrics,
   }
 }
 
+void BenchmarkExecutor::computeAveragePathSimilarities(
+    PlannerBenchmarkData& planner_data, const std::vector<planning_interface::MotionPlanDetailedResponse>& responses,
+    const std::vector<bool>& solved)
+{
+  ROS_INFO("Computing result path similarity");
+  const size_t result_count = planner_data.size();
+  size_t unsolved = std::count_if(solved.begin(), solved.end(), [](bool s) { return !s; });
+  std::vector<double> average_distances(responses.size());
+  for (size_t first_traj_i = 0; first_traj_i < result_count; ++first_traj_i)
+  {
+    // If trajectory was not solved there is no valid average distance so it's set to max double only
+    if (!solved[first_traj_i])
+    {
+      average_distances[first_traj_i] = std::numeric_limits<double>::max();
+      continue;
+    }
+    // Iterate all result trajectories that haven't been compared yet
+    for (size_t second_traj_i = first_traj_i + 1; second_traj_i < result_count; ++second_traj_i)
+    {
+      // Ignore if other result has not been solved
+      if (!solved[second_traj_i])
+        continue;
+
+      // Get final trajectories
+      const robot_trajectory::RobotTrajectory& traj_first = *responses[first_traj_i].trajectory_.back();
+      const robot_trajectory::RobotTrajectory& traj_second = *responses[second_traj_i].trajectory_.back();
+
+      // Compute trajectory distance
+      double trajectory_distance;
+      if (!computeTrajectoryDistance(traj_first, traj_second, trajectory_distance))
+        continue;
+
+      // Add average distance to counters of both trajectories
+      average_distances[first_traj_i] += trajectory_distance;
+      average_distances[second_traj_i] += trajectory_distance;
+    }
+    // Normalize average distance by number of actual comparisons
+    average_distances[first_traj_i] /= result_count - unsolved - 1;
+  }
+
+  // Store results in planner_data
+  for (size_t i = 0; i < result_count; ++i)
+    planner_data[i]["average_waypoint_distance REAL"] = moveit::core::toString(average_distances[i]);
+}
+
+bool BenchmarkExecutor::computeTrajectoryDistance(const robot_trajectory::RobotTrajectory& traj_first,
```

Comment:
Personally, I don't like this approach to error handling very much. I would prefer to return the cost
and throw `std::invalid_argument` when trajectories are empty.
My suggestion is to move the empty check up to the caller.

Reply:
``invalid_argument`` is currently not used anywhere in the moveit codebase, according to grep. this change request is non-standard and scope creep from the focus of this PR, as moveit returns bool on success all the time.

Reply:
`std::invalid_argument` is the standard exception to throw if an argument is invalid - as the name suggests. The fact that MoveIt doesn't use such standards yet is actually a pity...
However, I don't want to raise such a discussion for this PR.